core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -137,6 +155,18 @@ public String getLoadBalancerUserFacingFQDN(Long stackId) {
                 .orElseGet(() -> findPrivateLbNameOrNull(loadBalancers));
     }
 
+    public Optional<LoadBalancer> getLoadBalancer(Long stackId, LoadBalancerType loadBalancerType) {
```

Comment:
Could we change this method name to `getLoadBalancerWithEndpointDefined` or something like that, to make it clear you're only getting a load balancer if the FQDN/DNS/IP are defined?